### PR TITLE
Fixed global-search keyboard shortcut for non-english keyboard layouts.

### LINKED ISSUE #10646: Using "/" to focus search box does not work in Firefox
Typing "/" in Firefox (german keyboard layout, but that shouldn't make a difference) does not focus the search box but rather brings up Firefox's builtin quick search. See https://support.mozilla.org/en-US/kb/keyboard-shortcuts-perform-firefox-tasks-quickly#w_search -- I am not sure if there is a good way to override it.

Nomad version is 1.1.0

### PR DESCRIPTION
**Pull request by apollo13**
Closes #10646

### CODE REVIEW

File: ui/app/components/global-search/control.js [hunk 1]
```diff
@@ -57,32 +57,40 @@ export default class GlobalSearchControl extends Component {
     const allTaskGroupResults = results.Matches.groups || [];
     const allCSIPluginResults = results.Matches.plugins || [];
 
-    const jobResults = allJobResults.slice(0, MAXIMUM_RESULTS).map(({ ID: name, Scope: [ namespace, id ]}) => ({
```
Comment (tgross) [inline ui/app/components/global-search/control.js:60]: Can we back out the style changes here?

**Review by tgross — CHANGES_REQUESTED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ BUG FIXES:
 * drivers/exec: Fixed a bug where `exec` and `java` tasks inherit the Nomad agent's `oom_score_adj` value [[GH-10698](https://github.com/hashicorp/nomad/issues/10698)]
 * quotas (Enterprise): Fixed a bug where stopped allocations for a failed deployment can be double-credited to quota limits, resulting in a quota limit bypass. [[GH-10694](https://github.com/hashicorp/nomad/issues/10694)]
 * ui: Fixed a bug where exec would not work across regions. [[GH-10539](https://github.com/hashicorp/nomad/issues/10539)]
+* ui: Fixed global-search shortcut for non-english keyboards. [[GH-10646](https://github.com/hashicorp/nomad/issues/10646)]
```
Comment (tgross) [inline CHANGELOG.md:20]: Our changelogs point to the pull request, not the issue:

```suggestion
* ui: Fixed global-search shortcut for non-english keyboards. [[GH-10714](https://github.com/hashicorp/nomad/issues/10714)]
```

**Review by tgross — APPROVED** [review]

**Review by tgross — CHANGES_REQUESTED** [review]
@apollo13 apparently this broke the tests for this feature (see https://app.circleci.com/pipelines/github/hashicorp/nomad/16480/workflows/7875e4f3-ede6-4c81-9eeb-5834becb6ea5/jobs/159133), so maybe there's some magic here around key codes that I'm not aware of.

**Comment by apollo13** [comment]
Thanks @tgross, I think the error makes sense since it sends `keyCode` as event data and not `key`. I added a commit to fix that. The error makes sense since manually triggering a event with only `keyCode` is not what the browser would do…

**Comment by apollo13** [comment]
I have no clue why the github workflow fails. But aside from that the patch looks good :)

**Comment by tgross** [comment]
> I have no clue why the github workflow fails.

That GitHub Action needs to write a comment in the PR... which needs a GH token with write permissions and we don't expose that to forks. Safe to ignore.

LGTM!

**Review by tgross — APPROVED** [review]

**Pull request merged**